samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -133,4 +139,12 @@ final void writePlanJsonFile(String planJson) {
       log.warn("Failed to write execution plan json to file", e);
     }
   }
+
+  private void validateConfig() {
+    String appClassName = new ApplicationConfig(config).getAppClass();
+    if (appClassName != null && !config.containsKey(TaskConfig.WINDOW_MS())) {
```

Comment:
Yeah, still not every job will be doing join or window, so it seems odd to force those who don't do those operations to configure something unnecessary. 

I think these validations could fit very nicely into the planner, which is the first place we take a collective look at the DAG